model-optimizer/mo/utils/ir_reader/extender.py

```diff
@@ -36,6 +37,17 @@ def attr_to_list(node: Node, attribute: str):
 
     @staticmethod
     def const_shape_infer(node: Node):
+        # Check equality of old (restored from IR) and new (calculated while shape inference) input shapes
+        node['new_input_shapes'] = list()
+        for n in node.in_nodes():
+            node.new_input_shapes.append(node.in_node(n).shape)
+        assert len(node.new_input_shapes) == len(node.old_input_shapes), \
+            'Something wrong happened while {} node with type {} copy shape inference!'.format(node.name, node.type)
+        for i in range(len(node.new_input_shapes)):
```

Comment:
`for new_input_shape, old_input_shape in zip(node.new_input_shapes, node.old_input_shapes):`

Reply:
Yes, thanks, this variant is more clear to understanding.